server/shared/src/main/scala/org/http4s/server/middleware/CORS.scala

```diff
@@ -194,10 +194,11 @@ object CORS {
     "Depends on a deficient `CORSConfig`. See https://github.com/http4s/http4s/security/advisories/GHSA-52cf-226f-rhr6. If config.anyOrigin is true and config.allowCredentials is true, then the `Access-Control-Allow-Credentials` header will be suppressed starting with 0.22.3.",
     "0.21.27",
   )
-  @nowarn("cat=deprecation")
+  @nowarn // silence deprecation warning on Scala 3
   def apply[F[_], G[_]](http: Http[F, G], config: CORSConfig = CORSConfig.default)(implicit
       F: Applicative[F]
   ): Http[F, G] = {
+    val purposelyUnused = 0 // to satisfy @nowarn on Scala 2
```

Comment:
Enabling fatal warnings on both Scala 2 and Scala 3 just got a lot more tenuous. cc @satorg

Reply:
Do you mean, `@nowarn("cat=deprecation")` does not work on Scala3?
Because to me it seems it does:
```scala
import scala.annotation.nowarn

object DeprecationTest {

  @deprecated("this has been deprecated", "now")
  def deprecatedMethod(): Unit = ???

  @nowarn("cat=deprecation")
  def callDeprecatedMethod1(callDeprecated: Boolean): Unit = {
    if (callDeprecated)
      deprecatedMethod()
    else
      () // do nothing
  }

  def callDeprecatedMethod2(callDeprecated: Boolean): Unit = {
    if (callDeprecated)
      (deprecatedMethod(): @nowarn("cat=deprecation"))
    else
      () // do nothing
  }
}
```
Then, with help of scala-cli:
```
$ scala-cli compile -j 11 -S 2.12.17 -O -deprecation -- DeprecationTest.scala 
Compiling project (Scala 2.12.17, JVM)
Compiled project (Scala 2.12.17, JVM)

$ scala-cli compile -j 11 -S 2.13.10 -O -deprecation -- DeprecationTest.scala 
Compiling project (Scala 2.13.10, JVM)
Compiled project (Scala 2.13.10, JVM)

$ scala-cli compile -j 11 -S 3.1.3 -O -deprecation -- DeprecationTest.scala 
Compiling project (Scala 3.1.3, JVM)
Compiled project (Scala 3.1.3, JVM)

$ scala-cli compile -j 11 -S 3.2.0 -O -deprecation -- DeprecationTest.scala 
Compiling project (Scala 3.2.0, JVM)
Compiled project (Scala 3.2.0, JVM)
```
I.e. it seems it actually suppresses the deprecation warning in all cases.

Reply:
@satorg thanks for looking at this. The problem was that `@nowarn("cat=deprecation")` was complaining on 2.13.10 that there was no warning to suppress. Meanwhile, Scala 3 had a deprecation warning there that did need suppression.

Reply:
@satorg also I think the issue was here
```
config: CORSConfig = CORSConfig.default
```

Specifically, that the default arg is deprecated.

Reply:
Ok I see. Seems it is yet another "improvement" in warnings handling that was introduced in Scala 2.13.9.
In fact, it started way better detecting whether `@nowarn` is necessary or not. 
But yeah, it also causes cross-building issues...

Reply:
@armanbilge would you take a look at #6751 please?